cmd/config/internal/commands/cmdcreatesetter.go

```diff
@@ -138,6 +148,99 @@ func (r *CreateSetterRunner) preRunE(c *cobra.Command, args []string) error {
 	return nil
 }
 
+func (r *CreateSetterRunner) processSchema() error {
+	sc, err := schemaFromFile(r.SchemaPath)
+	if err != nil {
+		return err
+	}
+
+	flagType := r.CreateSetter.Type
+	var schemaType string
+	switch {
+	// json schema allows more than one type to be specified, but openapi
+	// only allows one. So we follow the openapi convention.
+	case len(sc.Type) > 1:
+		return errors.Errorf("only one type is supported: %s",
+			strings.Join(sc.Type, ", "))
+	case len(sc.Type) == 1:
+		schemaType = sc.Type[0]
+	}
+
+	// Since type can be set both through the schema file and through the
+	// --type flag, we make sure the same value is set in both places. If they
+	// are both set with different values, we return an error.
+	switch {
+	case flagType == "" && schemaType != "":
+		r.CreateSetter.Type = schemaType
+	case flagType != "" && schemaType == "":
+		sc.Type = []string{flagType}
+	case flagType != "" && schemaType != "":
+		if flagType != schemaType {
+			return errors.Errorf("type provided in type flag (%s) and in schema (%s) doesn't match",
+				r.CreateSetter.Type, sc.Type[0])
+		}
+	}
+
+	// Validate that all the types set in the schema are valid types.
+	err = validateTypes(sc)
+	if err != nil {
+		return err
+	}
+
+	// Only marshal the properties in SchemaProps. This means any fields in
+	// the schema file that isn't recognized will be dropped.
+	b, err := json.Marshal(sc.SchemaProps)
+	if err != nil {
+		return errors.Errorf("error marshalling schema: %v", err)
+	}
+	r.CreateSetter.Schema = string(b)
+	return nil
+}
+
+// The types recognized by by the go openapi validation library:
+// https://github.com/go-openapi/validate/blob/master/helpers.go#L35
+var validTypeValues = []string{
```

Comment:
Should this check only be for type validations? Or should we check for other schema properties as well? Like max instead of maximum.

Reply:
Any typos in the keys in the schema will be dropped as they will not be unmarshaled into the `SchemaProps` object in the schema. I have left a TODO that we should see if we should return an error if there are any fields in the schema that doesn't end up in the `SchemaProps` object. It just needs some investigation to figure out how best to do this without having to check a large number of properties on the schema.